PurchasesCoreSwift/Networking/ETagManager.swift

```
@@ -0,0 +1,184 @@
+        if ((eTagInResponse == nil)) {
+            eTagInResponse = headersInResponse[ETAG_HEADER_NAME.lowercased()] as! String?
+        }
+        
+        if (eTagInResponse != nil) {
+            if (shouldUseCachedVersion(responseCode: statusCode)) {
+                guard let storedResponse = getStoredHTTPResponse(for: request) else {
+                    if (retried) {
+                        Logger.warn(
+                                String(
+                                        format: Strings.network.could_not_find_cached_response_in_already_retried,
+                                        resultFromBackend
+                                )
+                        )
+                        return resultFromBackend
+                    } else {
+                        return nil
+                    }
+                }
+                return storedResponse
+            }
+            storeStatusCodeAndResponseIfNoError(
+                    for: request,
+                    statusCode: statusCode,
+                    responseObject: responseObject,
+                    eTag: eTagInResponse!)
+        }
+
+        return resultFromBackend
+    }
+
+    @objc public func clearCaches() {
+        if let bundleID = Bundle.main.bundleIdentifier {
+            self.userDefaults.removePersistentDomain(forName: bundleID + ".revenuecat.etags")
+        }
+    }
+
+    private func shouldUseCachedVersion(responseCode: Int) -> Bool {
+        responseCode == HTTPStatusCodes.notModifiedResponseCode.rawValue
+    }
+
+    private func getStoredETagAndResponse(for request: URLRequest) -> ETagAndResponseWrapper? {
+        if let cacheKey = eTagDefaultCacheKey(for: request) {
+            if let data = userDefaults.object(forKey: cacheKey) as! Data? {
+                do {
+                    return try ETagAndResponseWrapper(with: data)
+                } catch {
+                }
```

what's happening with this empty catch? when could it happen?

---

if we want to just return nil in case of error, then this is equivalent to the do try catch block: 

```swift
return try? ETagAndResponseWrapper(with: data)
```

---

Yes so the constructor of ETagAndResponseWrapper can throw if there's a problem with the JSONSerialization.jsonObject. That's why the try clause. I like your solution better.